src/chessboard/components/Arrows.tsx

```diff
@@ -40,9 +40,24 @@ export const Arrows = () => {
           boardWidth,
           arrowEndField
         );
-        let ARROW_LENGTH_REDUCER = boardWidth / 32;
+
+        const dx = to.x - from.x;
+        const dy = to.y - from.y;
+        const r = Math.hypot(dy, dx);
 
         const isArrowActive = i === arrows.length;
+
+        // Knight move is where the move is a 2:1 ratio, but to limit it to legal knight moves
+        // we check the difference between the start and end rank is 2 or less
+        const isKnightMove =
+          (Math.abs(dx) === 2 * Math.abs(dy) ||
+            Math.abs(dy) === 2 * Math.abs(dx)) &&
+          Math.abs(
+            parseInt(arrowStartField.split("")[1]) -
+              parseInt(arrowEndField.split("")[1])
+          ) <= 2;
```

Comment:
I guess this check will be enough 
```suggestion
        const isKnightMove = r === Math.hypot(1,2)

Reply:
Doesn't work as r is already the hypot of dx and dy which is usually in the hundreds, and the hypot of 1 & 2 is 2.236... - So you'd have to somehow scale down r (or 1,2 up) to get it to work. Would definitely like to find a more elegant solution though, because I accept mine is a bit chaotic!

Reply:
https://github.com/Clariity/react-chessboard/assets/42891870/9769b488-2449-4387-864d-bae95309a4f7

Reply:
Aaah with the addition of the * squareWidth yes. Very elegant!